### PR TITLE
docs: update Thread CoAP client MED/SED info

### PR DESCRIPTION
**Pull request by greg-fer**
Changed SED to MED in Thread CoAP client sample docs.
KRKNWK-7186

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: samples/openthread/coap_client/README.rst [hunk 1]
```diff
@@ -119,15 +119,16 @@ After building the sample and programming it to your development kit, test it by
 Testing Minimal Thread Device
 -----------------------------
 
-After building the MTD variant of this sample and programming it, the device starts in the SED mode with LED 3 disabled.
-You can switch to the MED mode at any moment during the standard testing procedure.
+After building the MTD variant of this sample and programming it, the device starts in the MED mode with **LED 3** on.
+This means that the radio is enabled, but the serial console is disabled to save power.
```
Comment (pszkotak) [inline samples/openthread/coap_client/README.rst:123]:
```suggestion
This means that the radio is enabled when idle and the serial console is operating.
```

File: samples/openthread/coap_client/README.rst [hunk 2]
```diff
@@ -119,15 +119,16 @@ After building the sample and programming it to your development kit, test it by
...
+You can switch to the SED mode at any moment during the standard testing procedure.
 
-To toggle MED, press **Button 3** on the client node.
-**LED 3** turns on to indicate the switch to the MED mode.
-At this point, the radio is enabled and power consumption increases.
+To toggle SED, press **Button 3** on the client node.
+**LED 3** turns off to indicate the switch to the SED mode.
+At this point, the radio is disabled and power consumption decreases.
```
Comment (pszkotak) [inline samples/openthread/coap_client/README.rst:128]:
```suggestion
At this point, the radio is disabled when idle and the serial console is not working to decrease the power consumption.
```

File: samples/openthread/coap_client/README.rst [hunk 3]
```diff
@@ -119,15 +119,16 @@ After building the sample and programming it to your development kit, test it by
...
 
-Pressing **Button 3** again will switch the mode back to SED.
-This does not affect the standard testing procedure.
+Pressing **Button 3** again will switch the mode back to MED.
+Switching between SED and MED modes does not affect the standard testing procedure.
```
Comment (pszkotak) [inline samples/openthread/coap_client/README.rst:131]: Not sure if we should expand this part, leaving it to your judgement.
```suggestion
Switching between SED and MED modes does not affect the standard testing procedure, except for the fact that LEDs and terminal logs are not available in the SED mode.
```

**Review by pszkotak — CHANGES_REQUESTED** [review]
Some clarifications needed.
Please note that in the MED mode there is a serial console avaialble.

File: samples/openthread/coap_client/README.rst (hunk 3 above)
Comment (pszkotak) [reply samples/openthread/coap_client/README.rst:131]: My bad, LEDs work just fine in the SED mode. Only the LED3 is turned off.

**Review by pszkotak — APPROVED** [review]
LGTM!

**Review by rlubos — APPROVED** [review]

**Pull request merged**